docs/content/docs/microsoft-agent-framework/auth.mdx

```diff
@@ -0,0 +1,131 @@
+---
+title: "Authentication"
+icon: "lucide/Shield"
+description: "Secure your Microsoft Agent Framework agents with user authentication"
+---
+
+## Overview
+
+CopilotKit supports forwarding user authentication to your Microsoft Agent Framework AG-UI server. The typical pattern is:
+
+- **Frontend**: Pass an auth token via `<CopilotKit properties={{ authorization: token }}>`.
+- **Backend (ASP.NET Core)**: Validate the token (for example, using Microsoft Entra ID) and scope the request accordingly.
+
+## How It Works
+
+```mermaid
+sequenceDiagram
+    participant Frontend
+    participant CopilotKit
+    participant AGUIServer
+    participant Agent
+
+    Frontend->>CopilotKit: authorization: "user-token"
+    CopilotKit->>AGUIServer: Forward auth token
+    AGUIServer->>Agent: user info available to your logic
+    Agent->>Agent: Access authenticated user context
+```
+
+## Frontend setup
+
+Pass your authentication token via the `properties` prop:
+
+```tsx
+<CopilotKit
+  runtimeUrl="/api/copilotkit"
+  properties={{
+    authorization: userToken, // Forwarded as Bearer token
+  }}
+>
+  <YourApp />
+</CopilotKit>
+```
+
+The `authorization` property is forwarded to your AG-UI server. If your server expects a Bearer token, include the `Bearer ` prefix when validating.
+## ASP.NET Core AG-UI server
+
+Validate the incoming token and scope requests in your AG-UI server. Below is a minimal example that reads the forwarded `authorization` value and performs custom validation before creating the agent.
+
+```csharp title="Program.cs"
+using Azure.AI.OpenAI;
+using Azure.Identity;
+using Microsoft.Agents.AI;
+using Microsoft.Agents.AI.Hosting.AGUI.AspNetCore;
+
+var builder = WebApplication.CreateBuilder(args);
+builder.Services.AddHttpClient().AddLogging();
+var app = builder.Build();
+
+string endpoint = builder.Configuration["AZURE_OPENAI_ENDPOINT"]!;
+string deployment = builder.Configuration["AZURE_OPENAI_DEPLOYMENT_NAME"]!;
+
+app.MapAGUIAgent("/", (messages, tools, context, forwardedProps) =>
+{
+    var authHeader = forwardedProps?.GetValueOrDefault("authorization");
+    if (string.IsNullOrEmpty(authHeader) || !authHeader.StartsWith("Bearer ", StringComparison.OrdinalIgnoreCase))
+    {
+        throw new UnauthorizedAccessException("Unauthorized");
+    }
+
+    var token = authHeader["Bearer ".Length..];
+    var user = ValidateYourToken(token); // implement your validation here
```

Comment:
Follow the guidance https://learn.microsoft.com/en-us/aspnet/core/security/authentication/configure-jwt-bearer-authentication?view=aspnetcore-9.0#jwt-bearer-token-basic-validation for auth.